src/navigation/card/screens/CardPairingScreen.tsx

```diff
@@ -5,52 +5,89 @@ import {VirtualDesignCurrency} from '../../../store/card/card.types';
 import {useAppDispatch} from '../../../utils/hooks';
 import BasePairing from '../../bitpay-id/components/BasePairing';
 import {CardScreens, CardStackParamList} from '../CardStack';
+import {incomingData} from '../../../store/scan/scan.effects';
+import {StackActions, useNavigation} from '@react-navigation/native';
 
 export type CardPairingScreenParamList =
   | {
       secret?: string;
       code?: string;
       dashboardRedirect?: string;
       vcd?: VirtualDesignCurrency;
+      paymentUrl?: string;
     }
   | undefined;
 
 const CardPairingScreen: React.FC<
   StackScreenProps<CardStackParamList, CardScreens.PAIRING>
 > = props => {
-  const {navigation, route} = props;
-  const {secret, code} = route.params || {};
-
+  const {route} = props;
+  const {secret, code, paymentUrl, dashboardRedirect} = route.params || {};
+  const navigation = useNavigation();
   const dispatch = useAppDispatch();
 
   const onSuccess = useCallback(() => {
-    const virtualDesignCurrency = route.params?.vcd as
-      | VirtualDesignCurrency
-      | undefined;
+    if (dashboardRedirect) {
+      const virtualDesignCurrency = route.params?.vcd as
+        | VirtualDesignCurrency
+        | undefined;
+
+      dispatch(
+        CardActions.virtualDesignCurrencyUpdated(
+          virtualDesignCurrency || 'bitpay-b',
+        ),
+      );
+      return;
+    }
 
-    dispatch(
-      CardActions.virtualDesignCurrencyUpdated(
-        virtualDesignCurrency || 'bitpay-b',
-      ),
-    );
-  }, [dispatch, route.params?.vcd]);
+    if (paymentUrl) {
+      navigation.dispatch(StackActions.replace('Tabs', {screen: 'Home'}));
```

Comment:
Since Home is always open, I feel like performance would be better by just navigating instead of replacing. Otherwise it seems like it's unmounting everything then remounting Tabs/Home again. Maybe by doing a check to see if Tabs/Home exists in the stack somewhere first, else do a replace?